browser/ui/webui/new_tab_page/brave_new_tab_message_handler.cc

```diff
@@ -558,13 +558,21 @@ void BraveNewTabMessageHandler::HandleGetWallpaperData(
       data->FindBool(ntp_background_images::kIsBackgroundKey);
   DCHECK(is_background);
 
+  constexpr char kBackgroundWallpaperKey[] = "backgroundWallpaper";
   if (is_background.value()) {
-    constexpr char kBackgroundWallpaperKey[] = "backgroundWallpaper";
     wallpaper.Set(kBackgroundWallpaperKey, std::move(*data));
     ResolveJavascriptCallback(args[0], base::Value(std::move(wallpaper)));
     return;
   }
 
+  // Even though we show sponsored image, we should pass "Background wallpaper"
+  // data so that NTP can know which wallpaper is selected by users.
```

Comment:
nit: `NTP customization menu` would be better for understanding instead of `NTP`?